lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -77,6 +90,14 @@ def define_virtual_aggregate_attribute(name, relation, method_name, column, opti
         end
       end
 
+      %w[average maximum minimum].each do |m_name|
+        define_method("define_virtual_#{m_name}_method") do |name, relation, column|
+          define_method(name) do
+            (has_attribute?(name) ? self[name] : send(relation).try(m_name.to_sym, column)) || 0
+          end
+        end
+      end
+
       def define_virtual_size_method(name, relation)
         define_method(name) do
           (has_attribute?(name) ? self[name] : send(relation).try(:size)) || 0
```

Comment:
this works because relation (that is loaded).size is cached locally

yea, so this is here when the relation returns a nil - which is totally invalid.
Probably in the case where the relation is a virtual aggregate to a parent manager and the parent manager is not populated.

Should totally find another way to solve that problem

Reply:
this is individual because the interface to determine this value is simply `relation.size` here

but for the others it is `relation.maximum(column)`

Average may be slightly different so we may want to handle that in its own method.
That or allow a block to be passed in